### DISCUSSION TITLE
Is it possible to flash the OEM WIFI Stick, so that this Stick is getting the powerdemand from a 3em and sending the power demand to the soyosource

### DISCUSSION THREAD

**Discussion by Rubber1Duck** [syssi/esphome-soyosource-gtn-virtual-meter#168] [Q&A] [answered]
Hello, as in the headline:
is it possible to flash the OEM WIFI Stick and this Stick than control the power demand witch is requested from a shelly 3em?
Tried some things but i am not able to establish this.
can someone provide a working example?
Thanks in advance
Peter

**Comment by syssi** [comment, marked as answer]
Yes. This is possible but a RS485 converter module must be attached to some free GPIOs. What did you try already?

**Reply by Rubber1Duck** [reply]
Hello Sebastian,
i tried this config "esp8266-wifi-dongle-limiter-example.yaml" (with imported "shelly-3em-http-status-json.yaml")  and i thougth this will be a version witch will do the job without any further equipment!? It was not clear for me that this will not work!
Does that mean that the "limiter" is not reachable over the USB-TTL port of the WIFI-Stick and it is not able to write a new limit?
okay i think  i now understand the "esp8266-wifi-dongle-limiter-example.yaml" the RS485 converter musst be connectet to this 2 pins (plus vcc and gnd of course)   tx_pin_rs485: GPIO5  rx_pin_rs485: GPIO14, correct?
oh man so manches mal hat man echt nen Brett vorm kopf .....................

**Reply by syssi** [reply]
Correct. The USB-TTL connection is used to retrieve some measurements and set some config settings. To control the power output of the inverter the RS485 limiter port must be used.

**Reply by Rubber1Duck** [reply]
thanks a lot, particularly for your great work and support. In the meanwile i found this
https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/discussions/96
and your answer in this
https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/discussions/155
okay lets order some RS485-TTL Adapter :-)
